mysql-test/main/partition_innodb.test

```diff
@@ -237,12 +237,11 @@ CREATE TABLE t1 (
 
 ALTER TABLE t1 PARTITION BY HASH (id) PARTITIONS 1;
 
---error ER_FEATURE_NOT_SUPPORTED_WITH_PARTITIONING
 ALTER TABLE t1 ADD CONSTRAINT test_ibfk_1 FOREIGN KEY (parent_id) REFERENCES t2 (id);
 
 ALTER TABLE t1 PARTITION BY HASH (id) PARTITIONS 2;
 
---error ER_FEATURE_NOT_SUPPORTED_WITH_PARTITIONING
+--error ER_CANT_CREATE_TABLE
 ALTER TABLE t1 ADD CONSTRAINT test_ibfk_1 FOREIGN KEY (parent_id) REFERENCES t2 (id);
 
 DROP TABLE t1, t2;
```

Comment:
Some functional testing with DML statements is needed here to demonstrate that the `FOREIGN KEY` constraint works after each of these `ALTER TABLE` operation, including the rejected one.

Please also add a comment here that there are further tests about this in `parts.foreign`. Initially, I was going to ask for this to be extended to cover `ON (UPDATE|DELETE) (CASCADE|SET NULL)`, but those are covered in that test.